src/components/edit_alarm_definition.ts

```diff
@@ -72,10 +77,46 @@ export class EditAlarmDefinitionPageCtrl {
     this.newAlarmDefinition.match_by.splice(index, 1);
   }
 
+  private _suggestAlarmActions(query, callback) {
+    return Promise.resolve(
+      this.notificationMethods.map(notification => notification.name)
+    ).then(callback);
+  }
+
+  private loadNotificationMethods() {
+    return this.monascaClientSrv
+      .listNotifications()
+      .then(
+        notification_methods =>
+          (this.notificationMethods = notification_methods)
+      );
+  }
+
+  public addAlarmAction() {
+    if (!this.newAlarmDefinition.alarm_actions_by_name) {
+      this.newAlarmDefinition.alarm_actions_by_name = [];
+    }
+    this.newAlarmDefinition.alarm_actions_by_name.push("");
+  }
+
+  public removeAlarmAction(index) {
+    if (!this.newAlarmDefinition.alarm_actions_by_name) {
+      return;
+    }
+    this.newAlarmDefinition.alarm_actions_by_name.splice(index, 1);
+  }
+
   // Edit Alarm Definition
   public saveAlarmDefinition() {
     this.saving = true;
     if (this.id) {
+      this.newAlarmDefinition.alarm_actions = this.newAlarmDefinition.alarm_actions_by_name.map(
+        alarm_action_name =>
+          this.notificationMethods.find(
+            notification_method => notification_method.name == alarm_action_name
+          ).id
+      );
+      delete this.newAlarmDefinition.alarm_actions_by_name;
```

Comment:
What's the reason for deleting this here?

Reply:
alarm_actions_by _name is just a temporary field used during manipulation, when the alarm is to be saved (ie. posted to monasca) that field is invalid and should be deleted